### PR TITLE
Allow tags update via Set operations for VirtualNetworkGateway and Vi…

### PR DESCRIPTION
**Pull request by ritwikbasu**
…rtualNetworkGatewayConnection

## Description

Allow tags update via Set-AzVirtualNetworkGateway and Set-AzVirtualNetworkGatewayConnection.

Commandlet review here - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/280

## Checklist

- [Y] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ Y] The title of the PR is clear and informative
- [ Y] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ Y] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ Y] If applicable, the changes made in the PR have proper test coverage
- [ Y] For public API changes to cmdlets:
    - [Y ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ Y] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
@ritwikbasu Please fix merge conflicts.

**Comment by ritwikbasu** [comment]
@maddieclayton @MiYanni @MikhailTryakhov - Resolved merge conflicts.

File: src/Network/Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs [hunk 1]
```diff
@@ -32,19 +33,12 @@ public class SetAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
...
-        [Parameter(
-            Mandatory = true,
-            ParameterSetName = VirtualNetworkGatewayParameterSets.Default,
             ValueFromPipeline = true,
             HelpMessage = "The virtual network gateway object to base modifications off of. This can be retrieved using Get-AzVirtualNetworkGateway")]
         public PSVirtualNetworkGateway VirtualNetworkGateway { get; set; }
 
         [Parameter(
             Mandatory = false,
-            ValueFromPipelineByPropertyName = true,
```
[the remark this answers is not in the document]
Comment (ritwikbasu) [reply src/Network/Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs]: Fixed

File: src/Network/Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs [hunk 2]
```diff
@@ -107,7 +102,6 @@ public class SetAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
...
```
Comment (MiYanni) [inline src/Network/Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs]: Missed one case where this change was not reverted.

Comment (ritwikbasu) [reply src/Network/Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs]: Fixed

**Pull request merged**